### DISCUSSION TITLE
Anisotropic diffusion coefficient

### DISCUSSION THREAD

**Discussion by Yeop-Jung** [kinnala/scikit-fem#923] [Q&A] [answered]
Hi, thanks for this nice library. 
I've been working on a Poisson equation with different diffusion coefficients by directions.
I believe it should be somehow included in a 2nd order tensor form.
Something like shown here;
https://fenicsproject.org/olddocs/dolfin/1.5.0/python/demo/documented/tensor-weighted-poisson/python/documentation.html

But I'm not sure how it's done in skfem. 
Is there any example of this?

**Comment by kinnala** [comment, marked as answer]
There is no example AFAIK, but it should be something along the following lines:
```python
from skfem.helpers import *

@BilinearForm
def anipoisson(u, v, w):
    C = np.array([[1.0, 0.1],
                  [0.1, 1.0]])
    return dot(mul(C, grad(u)), grad(v))
```

**Comment by Yeop-Jung** [comment]
Oh yes, 
This works with tutorial 1 just fine.
Thanks!

![image](https://user-images.githubusercontent.com/112521070/191324612-e84c44e6-cec0-427c-9b48-339e2edff1d3.png)

**Comment by Yeop-Jung** [comment]
@kinnala 
A subsequent question I have is then, how would you use the 2D diffusion tensor for the nonlinear Poisson case?
For example, when solving the problem in the "Discrete functions in forms" of How-to-guides,
![image](https://user-images.githubusercontent.com/112521070/191850019-bfcf59e0-ae1f-4dd3-b8cc-1eafd39949da.png)

It uses "w.u_k"  that's from  "basis.interpolate(x)"
```
@fem.BilinearForm
def bilinf(u, v, w):
    return (w.u_k + .1) * dot(grad(u), grad(v))

m = fem.MeshTri().refined(3)
basis = fem.Basis(m, fem.ElementTriP1())
b = unit_load.assemble(basis)
x = 0. * b.copy()
for itr in range(20):  # fixed point iteration
     A = bilinf.assemble(basis, u_k=basis.interpolate(x))
     x = fem.solve(*fem.condense(A, b, I=m.interior_nodes()))
     print(round(x.max(), 10))
```
I think this only works when it's 1D. But what if I want to use a 2D tensor with a nonlinear term something like this.?
np.array([[u_k1, 0], [0, u_k2]])  

Thanks!

**Reply by kinnala** [reply]
What you wrote should work after you replace the offdiagonal 0 by np.zeros_like(u_k1). If you have a specific PDE I can give an example.
